pkg/client/client_venconn.go

```diff
@@ -0,0 +1,215 @@
+	"github.com/jetstack/venafi-connection-lib/venafi_client/auth"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	ctrlruntimelog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type VenConnClient struct {
+	agentMetadata *api.AgentMetadata
+	connHandler   venafi_client.ConnectionHandler
+	installNS     string       // Namespace in which the agent is running in.
+	venConnName   string       // Name of the VenafiConnection resource to use.
+	venConnNS     string       // Namespace of the VenafiConnection resource to use.
+	client        *http.Client // Used to make HTTP requests to Venafi Cloud.
+}
+
+// NewVenConnClient lets you make requests to the Venafi Cloud backend using the
+// given VenafiConnection resource.
+//
+// You need to call Start to start watching the VenafiConnection resource. If
+// you don't, the client will be unable to find the VenafiConnection that you
+// are referring to as its client-go cache will remain empty.
+//
+// The http.Client is used for Venafi and Vault, not for Kubernetes. The
+// `installNS` is the namespace in which the agent is running in. The passed
+// `restcfg` is not mutated. `trustedCAs` is only used for connecting to Venafi
+// Cloud and Vault and can be left nil.
+func NewVenConnClient(restcfg *rest.Config, agentMetadata *api.AgentMetadata, installNS, venConnName, venConnNS string, trustedCAs *x509.CertPool) (*VenConnClient, error) {
+	// TODO(mael): The rest of the codebase uses the standard "log" package,
+	// venafi-connection-lib uses "go-logr/logr", and client-go uses "klog". We
+	// should standardize on one of them, probably "slog".
+	ctrlruntimelog.SetLogger(logr.Logger{})
+
+	if installNS == "" {
+		return nil, errors.New("programmer mistake: installNS must be provided")
+	}
+	if venConnName == "" {
+		return nil, errors.New("programmer mistake: venConnName must be provided")
+	}
+	if venConnNS == "" {
+		return nil, errors.New("programmer mistake: venConnNS must be provided")
+	}
+
+	restcfg = rest.CopyConfig(restcfg)
+	restcfg.Impersonate = rest.ImpersonationConfig{
+		UserName: fmt.Sprintf("system:serviceaccount:%s:venafi-connection", installNS),
+	}
+
+	// TLS-related configuration such as root CAs and client certs are contained
+	// in the restcfg; let's create an http.Client that uses them.
+	httpCl, err := rest.HTTPClientFor(restcfg)
+	if err != nil {
+		return nil, fmt.Errorf("while turning the REST config into an HTTP client: %w", err)
+	}
+
+	restMapper, err := apiutil.NewDynamicRESTMapper(restcfg, httpCl)
+	if err != nil {
+		return nil, fmt.Errorf("while creating the REST mapper: %w", err)
+	}
+
+	// This Kubernetes client only needs to be able to read and write the
+	// VenafiConnection resources and read Secret resources.
+	scheme := runtime.NewScheme()
+	_ = venapi.AddToScheme(scheme)
+	_ = corev1.AddToScheme(scheme)
```

Comment:
Why discard these errors?

Reply:
I discarded these errors because the tests would fail if these two schemas hadn't loaded properly. I guess I saved myself a few keystrokes... I don't think this will bite me later